services/session.ts

```diff
@@ -1,16 +1,38 @@
-// TODO: save users session in redis or database
-type UserType = { id: string; conversationId: string };
-const users: UserType[] = [];
-
-// Function to set user data in the session
-export function setUserSession(user: UserType) {
-  users.push(user);
-  return;
+import { z } from "zod";
+
+import { config } from "../utils/config";
+import inMemory from "../utils/session/in-memory";
+import redis from "../utils/session/redis";
+
+const schema = z.object({
+  waId: z.string(),
+  conversationId: z.string(),
+});
+
+type Session = z.input<typeof schema>;
+
+export async function setSession(session: Session) {
+  if (config.SESSION_DATABASE === "in-memory") {
+    await inMemory.setItem(session.waId, JSON.stringify(session));
```

Comment:
IMO to make is simpler, in-memory should just use variable like before, no need of `unstorage`